src/App.tsx

```diff
@@ -1,26 +1,226 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { changeTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem';
+import classNames from 'classnames';
+import { ErrorMessage } from './types/ErrorMessage';
+import { Form } from './components/Form';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [activeTodos, setActiveTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>('');
+  const [activeLink, setActiveLink] = useState('all');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const updateTodos = useCallback(
+    (link = activeLink) => {
+      switch (link) {
+        case 'all':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos);
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'));
+
+          break;
+        case 'active':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos.filter(todo => !todo.completed));
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'));
+
+          break;
+        case 'completed':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos.filter(todo => todo.completed));
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'));
+
+          break;
+      }
+    },
+    [activeLink],
+  );
+
+  const timeoutRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    if (errorMessage) {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+
+      timeoutRef.current = window.setTimeout(() => {
```

Comment:
```suggestion
      timeoutRef.current =setTimeout(() => {
```

nit: setTimout is accessible without `window.`

Reply:
Not fixed